app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx

```diff
@@ -0,0 +1,117 @@
+/** @file Main App module responsible for rendering virtual router. */
+
+import * as React from "react";
+import * as toast from "react-hot-toast";
+import * as router from "react-router-dom";
+
+import * as authProvider from "../authentication/providers/auth";
+import DashboardContainer from "../dashboard/components/dashboard";
+import * as authService from "../authentication/service";
+import withRouter from "../navigation";
+import * as loggerProvider from "../providers/logger";
+import * as session from "../authentication/providers/session";
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/** Path to the root of the app (i.e., the Cloud dashboard). */
+export const DASHBOARD_PATH = "/";
+
+
+
+// ===========
+// === App ===
+// ===========
+
+/** Global configuration for the `App` component. */
+export interface AppProps {
+    /** Logger to use for logging. */
+    logger: loggerProvider.Logger;
+    /** Whether the application is running on a desktop (i.e., versus in the Cloud). */
+    runningOnDesktop: boolean;
```

Comment:
consider `enum Platform { desktop = "desktop", cloud = "cloud" }` or `type Platform = "desktop" | "cloud"` to make it more self-documenting

Reply:
I like this suggestion, I'll add it.